apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTrackerLogic/process-tracker-adapters/third-party.proces-tracker.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  pluginsWhiteList,
+  processStatusToIcon,
+} from '@/common/components/molecules/ProcessTracker/constants';
+import {
+  IProcessTracker,
+  ProcessTrackerItem,
+} from '@/common/components/molecules/ProcessTracker/hooks/useProcessTrackerLogic/process-tracker-adapters/process-tracker.abstract';
+import { TPlugin } from '@/domains/workflow-definitions/fetchers';
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { ProcessStatus } from '@ballerine/common';
+
+export class ThirdPartyProcessTracker implements IProcessTracker {
+  PROCESS_NAME = 'third-party';
+
+  constructor(readonly workflow: TWorkflowById, readonly plugins?: TPlugin[]) {}
+
+  buildItems(): ProcessTrackerItem[] {
+    return (
+      this.plugins
+        ?.filter(({ name }) => pluginsWhiteList.includes(name as (typeof pluginsWhiteList)[number]))
+        ?.map(({ displayName, name }) => {
+          const pluginStatus = this.getPluginByName(name)?.status ?? ProcessStatus.DEFAULT;
+
+          return {
+            text:
+              pluginStatus === ProcessStatus.CANCELED ? (
+                <span className={`text-slate-400/40 line-through`}>{displayName}</span>
+              ) : (
+                displayName
+              ),
+            leftIcon: processStatusToIcon[pluginStatus as keyof typeof processStatusToIcon],
+          };
+        }) || []
+    );
+  }
+
+  getReadableName(): string {
+    return '3rd party processes';
+  }
+
+  private getPluginByName(name: string) {
+    let plugin: NonNullable<TWorkflowById['context']['pluginsOutput']>[string];
+
+    Object.keys(this.workflow?.context?.pluginsOutput ?? {})?.forEach(key => {
```

Comment:
Whenever you use .forEach you will run o(n) iterations even you found your relevant item,

change to :
```
Object.values(this.workflow?.context?.pluginsOutput ?? {}).find(
  (output) => output?.name === name
);
```

Reply:
This one is on me and I will fix it. Never occurred to me to use `Object.prototype.values` over `Object.prototype.keys`.